File: pptx-service/renderers/compare.py
from builder import add_bg, add_textbox, add_rect, add_eyebrow, add_line
from theme import ThemeColors


def render(slide, data: dict, t: ThemeColors) -> None:
    add_bg(slide, t.bg)
    heading = data.get("heading", "")
    left = data.get("left", {})
    right = data.get("right", {})
    eyebrow = data.get("eyebrow", "")

    if eyebrow:
        add_eyebrow(slide, eyebrow, t)
    add_textbox(slide, 0.83, 0.7, 11.0, 1.0, heading,
                t.font_display, 40, t.text, bold=True)

    add_line(slide, 6.67, 1.8, 6.67, 7.0, t.border, 1.0)

    for side, x_offset in [(left, 0.5), (right, 7.0)]:
        title = side.get("title", "")
        items = side.get("items", [])
        add_textbox(slide, x_offset, 1.9, 5.8, 0.8, title,
                    t.font_display, 32, t.accent, bold=True)
        y = 2.8
        for item in items[:6]:
            add_rect(slide, x_offset, y + 0.15, 0.08, 0.08, fill_hex=t.muted)
            add_textbox(slide, x_offset + 0.25, y, 5.4, 0.6, item,
                        t.font_body, 24, t.text, wrap=True)
            y += 0.68

File: lib/pptx/slides/compare.ts
import type { CompareSlide } from '@/lib/types';
import {
  px, py, SLIDE_W,
  setBackground, addRect, addLine, addTextBox, addHeading, addPageNumber,
  type PptxTheme,
} from '../helpers';
import type PptxGenJS from 'pptxgenjs';
type PSlide = ReturnType<InstanceType<typeof PptxGenJS>['addSlide']>;

export function render(slide: PSlide, data: CompareSlide, t: PptxTheme, n: number, total: number): void {
  setBackground(slide, '#' + t.bg);
  addHeading(slide, data.heading, t, py(60));

  const midX = SLIDE_W / 2;
  const colW = (SLIDE_W - 2 * t.paddingIn) / 2 - px(20);
  const leftX = t.paddingIn;
  const rightX = midX + px(20);
  const contentY = py(200);
  const contentH = py(760);

  addRect(slide, leftX, contentY, colW, contentH, '#' + t.rule, { transparency: 85 });
  addRect(slide, rightX, contentY, colW, contentH, '#' + t.accent, { transparency: 90 });
  addLine(slide, midX, contentY, midX, contentY + contentH, '#' + t.rule, 2);

  addTextBox(slide, data.left.title, {
    x: leftX + px(20), y: contentY + py(20), w: colW - px(40),
    fontSize: t.bodySize, color: '#' + t.text, fontFace: t.fontBody, bold: true,
  });
  data.left.items.forEach((item, i) => {
    addTextBox(slide, `• ${item}`, {
      x: leftX + px(20), y: contentY + py(100) + i * py(100), w: colW - px(40),
      fontSize: t.bodySize, color: '#' + t.text, fontFace: t.fontBody, wrap: true,
    });
  });

  addTextBox(slide, data.right.title, {
    x: rightX + px(20), y: contentY + py(20), w: colW - px(40),
    fontSize: t.bodySize, color: '#' + t.text, fontFace: t.fontBody, bold: true,
  });
  data.right.items.forEach((item, i) => {
    addTextBox(slide, `• ${item}`, {
      x: rightX + px(20), y: contentY + py(100) + i * py(100), w: colW - px(40),
      fontSize: t.bodySize, color: '#' + t.text, fontFace: t.fontBody, wrap: true,
    });
  });

  addPageNumber(slide, n, total, t);
}
